osgar/drivers/realsense.py

```diff
@@ -85,17 +89,25 @@ def pose_callback(self, frame):
     def depth_callback(self, frameset):
         try:
             assert frameset.is_frameset()
-            frame = frameset.as_frameset().get_depth_frame()
-            n = frame.get_frame_number()
+            depth_frame = frameset.as_frameset().get_depth_frame()
+            color_frame = frameset.as_frameset().get_color_frame()
```

Comment:
do we need color frame if we do not plan to publish RGB image?

Reply:
pyrealsense api is really really strange - I have no idea what happens when we request color frame but the respective stream was not enabled. It will most likely return something, that has `is_video_frame()` which returns `False`. That said, I would also prefer layout such as:

```python
if self.depth_rgb:
    color_frame = frameset.as_frameset().get_color_frame()
    assert color_frame.is_video_frame()
    color_image = np.asanyarray(color_frame.as_video_frame().get_data())
    _, data = cv2.imencode('*.jpeg', color_image)
   self.publish('color', data.tobytes())
```

as that keeps all the related stuff together.